From b705caef6c717adc80585843b7fcc68700ced4b6 Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Sat, 12 Aug 2017 09:25:24 -0700
Subject: [PATCH 1/4] Fix build with clang

Fix
error: logical not is only applied to the left hand side of this bitwise operator [-Werror,-Wlogical-not-parentheses]
        if ((!port_attr->comp_mask & IBV_EXP_QUERY_PORT_ATTR_MASK1) ||

Signed-off-by: Khem Raj <raj.khem@gmail.com>
---
Upstream-Status: Pending

 include/infiniband/verbs_exp.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/infiniband/verbs_exp.h b/include/infiniband/verbs_exp.h
index ae94deb..42ed83d 100644
--- a/include/infiniband/verbs_exp.h
+++ b/include/infiniband/verbs_exp.h
@@ -2955,7 +2955,7 @@ static inline int ibv_exp_query_port(struct ibv_context *context,
 				      &port_attr->port_attr);
 
 	/* Check that only valid flags were given */
-	if ((!port_attr->comp_mask & IBV_EXP_QUERY_PORT_ATTR_MASK1) ||
+	if (!(port_attr->comp_mask & IBV_EXP_QUERY_PORT_ATTR_MASK1) ||
 	    (port_attr->comp_mask & ~IBV_EXP_QUERY_PORT_ATTR_MASKS) ||
 	    (port_attr->mask1 & ~IBV_EXP_QUERY_PORT_MASK)) {
 		errno = EINVAL;
-- 
2.14.1

